### PR TITLE
adding subscribe to events

### PR DESCRIPTION
**Pull request by gijo-varghese**
Added a `subcribeEvents` prop which will receive a function. Events from the STM button will be sent to that function. 

e.g.:
```js
  eventListener = e => {
    console.log(e);
  };
  render() {
    return (
      <div>
        <MessengerSendToMessenger
          appId="893463920822463"
          ctaText={this.props.ctaText}
          pageId={this.props.pageId}
          size="xlarge"
          subscribeEvents={this.eventListener}
        />
      </div>
    );
  }
```

### CODE REVIEW

File: src/MessengerSendToMessenger.js [hunk 1]
```diff
@@ -39,6 +39,7 @@ export default class MessengerSendToMessenger extends Component {
     version: PropTypes.string,
     language: PropTypes.string,
     debug: PropTypes.bool,
+    subscribeEvents: PropTypes.func,
```
Comment (chentsulin) [inline src/MessengerSendToMessenger.js]: I prefer to name it `onSendToMessenger`.

File: src/MessengerSendToMessenger.js [hunk 2]
```diff
@@ -52,6 +53,7 @@ export default class MessengerSendToMessenger extends Component {
     version: '2.11',
     language: 'en_US',
     debug: false,
+    subscribeEvents: () => true,
```
Comment (chentsulin) [inline src/MessengerSendToMessenger.js]:
```js
onSendToMessenger: () => {},
```

File: src/MessengerSendToMessenger.js [hunk 3]
```diff
@@ -63,14 +65,23 @@ export default class MessengerSendToMessenger extends Component {
...
+      subscribeEvents,
+    } = this.props;
     window.fbAsyncInit = () => {
       window.FB.init({
         appId,
         autoLogAppEvents,
         xfbml,
         version: `v${version}`,
       });
+      window.FB.Event.subscribe('send_to_messenger', e => {
```
Comment (chentsulin) [inline src/MessengerSendToMessenger.js]:
```js
window.FB.Event.subscribe('send_to_messenger', event => {
  if (onSendToMessenger) onSendToMessenger(evemt);
});
```

**Review by chentsulin — CHANGES_REQUESTED** [review]

**Comment by gijo-varghese** [comment]
Ok. I'll update soon

**Comment by gijo-varghese** [comment]
I'll create another pull request

**Pull request closed without merging**